hank-core/src/main/java/com/liveramp/hank/storage/ReaderResult.java

```diff
@@ -63,12 +63,15 @@ public void found() {
 
   public void requiresBufferSize(int size) {
     if (buffer == null || buffer.capacity() < size) {
```

Comment:
I think maybe the logic could be a little simpler if this was structured if(buffer == null) {} else if(buffer.capacity() < size){}, but seems good to me either way.